Riot/Modules/Spaces/SpaceList/SpaceListViewModelType.swift

```diff
@@ -21,11 +21,13 @@ import Foundation
 protocol SpaceListViewModelViewDelegate: AnyObject {
     func spaceListViewModel(_ viewModel: SpaceListViewModelType, didUpdateViewState viewSate: SpaceListViewState)
     func spaceListViewModel(_ viewModel: SpaceListViewModelType, didSelectSpaceAt indexPath: IndexPath)
+    func spaceListViewModelRevertSelection(_ viewModel: SpaceListViewModelType)
```

Comment:
We should try to keep one entry point and one output point in the VM to manage view. Do not hesitate to create intermediate `struct` or change the view state data structure. The `enum` data structure is just a example.
```
struct SpaceListViewData {
 let sections: [SpaceListSection]
 let selectedIndexPath: IndexPath
 let isHomeSelected: Bool
 let refreshList: Bool // or create a view state `firstLoad(_ viewData: SpaceListViewData)`plus `loaded(_ viewData: SpaceListViewData)` is up to you
}
```
By updating the `loaded` view state or by creating new ones, you can get rid of these two delegates:
```
func spaceListViewModel(_ viewModel: SpaceListViewModelType, didSelectSpaceAt indexPath: IndexPath)
func spaceListViewModelRevertSelection(_ viewModel: SpaceListViewModelType)
```